src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.FruitDao;
+import core.basesyntax.db.FruitDaoImpl;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+    private final FruitDao fruitDao;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+        fruitDao = new FruitDaoImpl();
```

Comment:
According to the checklist, all services should be independent. Instantiating `FruitDaoImpl` inside `ShopServiceImpl` makes it dependent on a specific implementation of `FruitDao`. Consider injecting `FruitDao` through the constructor to follow the Dependency Inversion principle, part of the SOLID principles.